stripe-ui-core/src/main/java/com/stripe/android/uicore/elements/TextFieldUI.kt

```diff
@@ -267,6 +261,7 @@ internal fun TextFieldUi(
     shouldShowError: Boolean,
     modifier: Modifier = Modifier,
     visualTransformation: VisualTransformation = VisualTransformation.None,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
```

Comment:
How come this one is different?

Reply:
`MutableInteractionSource` isn't comparable so we add `remember` to keep this function stable, avoiding recompositions on every render.

Reply:
I was mostly curious why we need to provide one here at all.

Reply:
Ah my bad! This is to allow the `InteractionSource` to be attached to the actual `TextField` composable that we render.

```kotlin
@Composable
internal fun TextFieldUi(
    value: String,
    // ...
    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
    // ...
    onValueChange: (value: String) -> Unit = {},
    // ...
) {
    val colors = TextFieldColors(shouldShowError)

    TextField(
        value = value,
        onValueChange = onValueChange,
        // ....
        interactionSource = interactionSource,
        // ....
    )
}
```